subprojects/docs/src/docs/userguide/workingWithFiles.adoc

```diff
@@ -15,47 +15,288 @@
+
+----
+task packageDistribution(type: Zip) {
+    archiveName = "my-distribution.zip"
+    destinationDir = file("${buildDir}/dist")
+
+    from("${buildDir}/toArchive") {
+        exclude "**/*.pdf"
+    }
+
+    from("${buildDir}/toArchive") {
+        include "**/*.pdf"
+        into "docs"
+    }
+}
+----
+
+As you can see, you can have multiple `from()` declarations in a copy specification, each with its own configuration.
+
+[[sec:unpacking_archives_example]]
+=== Unpacking archives
+
+Archives are effectively self-contained file systems, so unpacking them is a case of copying the files from that file system into a directory (or even another archive). Gradle enables this by providing some wrapper functions that make archives available as hierarchical collections of files known as file trees (TODO: link to an in-depth discussion of file trees).
+
+The two functions of interest are api:org.gradle.api.Project#zipTree[] and api:org.gradle.api.Project#tarTree[], which produce a api:org.gradle.api.file.FileTree[] from a corresponding archive file. That file tree can then be used in a `from()` specification, like so:
+
+----
+task unpackFiles(type: Copy) {
+    from zipTree("src/resources/thirdPartyResources.zip")
+    into "${buildDir}/resources"
+}
+----
+
+As with a normal copy, you can control which files are unpacked via filters and even rename files as they are unpacked.
+
+=== Creating directories
+
+Many tasks need to create directories to store the files they generate, which is why Gradle automatically manages this aspect of tasks when they explicitly define file and directory outputs. You can learn about this feature in the <<sec:up_to_date_checks,incremental build>> section of the user guide. All core Gradle tasks ensure that any output directories they need are created if necessary using this mechanism.
+
+In cases where you need to create a directory manually, you can use the api:org.gradle.api.Project#mkdir[] method from within your build scripts or custom task implementations. Here's a simple example:
+
+----
+task ensureDirectory {
+    doLast {
+        mkdir "images"
+    }
+}
+----
+
+=== Moving files and directories
+
+Gradle has no API for moving files and directories around, but you can use the <<sec:ant,Apache Ant integration>> to easily do that, as shown in this example:
+
+----
+task moveReports {
+    doLast {
+        ant.move file: "${buildDir}/reports", todir: "${buildDir}/toArchive"
+    }
+}
+----
+
+This is not a common requirement and should be used sparingly as you lose information and can easily break a build. It's generally preferable to copy directories and files instead.
+
+=== Renaming files on copy
```

Comment:
One example on more complex processing would be great as well e.g. the use of `eachFile`.

Reply:
Is there a common scenario for users in which something like `eachFile()` is the best solution? Or is it only for specialised cases?